chalk-parse/src/ast.rs

```diff
@@ -41,6 +42,34 @@ pub struct StructFlags {
     pub fundamental: bool,
 }
 
+#[derive(Clone, PartialEq, Eq, Debug)]
+pub struct FnDefn {
+    pub name: Identifier,
+    pub parameter_kinds: Vec<ParameterKind>,
+    pub where_clauses: Vec<QuantifiedWhereClause>,
+    pub params: Vec<Field>,
+    pub returns: Vec<Field>,
+    pub flags: FnFlags,
+}
+
+#[derive(Clone, PartialEq, Eq, Debug)]
+pub struct FnFlags {
+    pub upstream: bool,
+    pub fundamental: bool,
+}
+
+#[derive(Clone, PartialEq, Eq, Debug)]
+pub struct ClosureDefn {
+    // What's a closure's name? Need to generate an `Identifier` or something
+    // TODO(markmccaskey): investigate what rustc does here
+    pub name: Identifier,
```

Comment:
Technically, closures aren't nameable. But I'm not sure if rustc generates a name.

Reply:
Yes, rustc has an internal name. I think giving them a dummy name is a fine thing to do for chalk tests.

Reply:
That said, we'll probably want to (a) consolidate some of this with functions and (b) we may want other things eventually to better match rustc. But we can evolve this part.